### PR TITLE
feat(jigasi): install openjdk nonheadless, currently required for dependencies

### PR DESCRIPTION
**Pull request by aaronkvanmeerten**

### CODE REVIEW

**Review by damencho — APPROVED** [review]

File: jigasi/Dockerfile [hunk 1]
```diff
@@ -11,8 +11,8 @@ LABEL org.opencontainers.image.documentation="https://jitsi.github.io/handbook/"
 ENV GOOGLE_APPLICATION_CREDENTIALS /config/key.json
 
 RUN apt-dpkg-wrap apt-get update && \
-    apt-dpkg-wrap apt-get install -y jigasi jq jitsi-autoscaler-sidecar && \
-    apt-cleanup
+apt-dpkg-wrap apt-get install -y openjdk-17-jre openjdk-17-jdk jigasi jq jitsi-autoscaler-sidecar && \
```
Comment (saghul) [inline jigasi/Dockerfile:14]: Style: pl keep the alignment.

Comment (saghul) [inline jigasi/Dockerfile:14]: Do we need the JDK for jvb or jibri? If not, can we drop it from base-java?

**Comment by saghul** [comment]
What dependencies need this?

File: jigasi/Dockerfile (hunk 1 above)
Comment (aaronkvanmeerten) [reply jigasi/Dockerfile:14]: not sure about JVB/jibri re: JDK but I do know without these we get the following exception for jigasi:

`Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.UnsatisfiedLinkError: Can't load library: /usr/lib/jvm/java-17-openjdk-arm64/lib/libawt_xawt.so [in thread "ForkJoinPool.commonPool-worker-2"]`

Comment (aaronkvanmeerten) [reply jigasi/Dockerfile:14]: my bad

**Review by aaronkvanmeerten — COMMENTED** [review]
> What dependencies need this? 

Full jigasi error when this isn't present:

2024-09-04 19:08:12.014 INFO: [18] LibJitsiImpl$ServiceLock.initializeService#208: Failed to initialize service implementation org.jitsi.impl.neomedia.MediaServiceImpl.
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at org.jitsi.impl.libjitsi.LibJitsiImpl$ServiceLock.initializeService(LibJitsiImpl.java:192)
	at org.jitsi.impl.libjitsi.LibJitsiImpl$ServiceLock.getService(LibJitsiImpl.java:131)
	at org.jitsi.impl.libjitsi.LibJitsiImpl.getService(LibJitsiImpl.java:91)
	at org.jitsi.impl.libjitsi.LibJitsiOSGiImpl.getService(LibJitsiOSGiImpl.java:69)
	at org.jitsi.service.libjitsi.LibJitsi.invokeGetServiceOnImpl(LibJitsi.java:170)
	at org.jitsi.service.libjitsi.LibJitsi.getMediaService(LibJitsi.java:122)
	at net.java.sip.communicator.impl.neomedia.NeomediaActivator.startWithServices(NeomediaActivator.java:417)
	at net.java.sip.communicator.util.osgi.DependentActivator.addingService(DependentActivator.java:103)
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:943)
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:871)
	at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	at org.osgi.util.tracker.AbstractTracked.trackInitial(AbstractTracked.java:183)
	at org.osgi.util.tracker.ServiceTracker.open(ServiceTracker.java:321)
	at org.osgi.util.tracker.ServiceTracker.open(ServiceTracker.java:264)
	at net.java.sip.communicator.util.osgi.DependentActivator.start(DependentActivator.java:79)
	at org.jitsi.impl.osgi.framework.BundleImpl.start(BundleImpl.java:428)
	at org.jitsi.impl.osgi.framework.launch.FrameworkImpl.startLevelChanged(FrameworkImpl.java:483)
	at org.jitsi.impl.osgi.framework.startlevel.FrameworkStartLevelImpl$Command.run(FrameworkStartLevelImpl.java:133)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class java.awt.Toolkit
	at java.desktop/java.awt.Dimension.<clinit>(Dimension.java:90)
	at org.jitsi.impl.neomedia.device.DeviceConfiguration.<clinit>(DeviceConfiguration.java:198)
	at org.jitsi.impl.neomedia.MediaServiceImpl.<init>(MediaServiceImpl.java:163)
	... 29 more
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.UnsatisfiedLinkError: Can't load library: /usr/lib/jvm/java-17-openjdk-arm64/lib/libawt_xawt.so [in thread "ForkJoinPool.commonPool-worker-2"]
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2398)
	at java.base/java.lang.Runtime.load0(Runtime.java:755)
	at java.base/java.lang.System.load(System.java:1957)
	at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
	at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:388)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:232)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:174)
	at java.base/jdk.internal.loader.NativeLibraries.findFromPaths(NativeLibraries.java:315)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:285)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2403)
	at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:818)
	at java.base/java.lang.System.loadLibrary(System.java:1993)
	at java.desktop/java.awt.Toolkit$2.run(Toolkit.java:1388)
	at java.desktop/java.awt.Toolkit$2.run(Toolkit.java:1386)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
	at java.desktop/java.awt.Toolkit.loadLibraries(Toolkit.java:1385)
	at java.desktop/java.awt.Toolkit.initStatic(Toolkit.java:1423)
	at java.desktop/java.awt.Toolkit.<clinit>(Toolkit.java:1397)
	at java.desktop/java.awt.Component.<clinit>(Component.java:624)
	at net.sf.fmj.media.renderer.video.SimpleSwingRenderer.<init>(SimpleSwingRenderer.java:213)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128)
	at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:347)
	at java.base/java.lang.Class.newInstance(Class.java:645)
	at net.sf.fmj.utility.PlugInUtility.getPlugInInfo(PlugInUtility.java:27)
	at net.sf.fmj.registry.Registry.setDefaults(Registry.java:394)
	at net.sf.fmj.registry.Registry.<init>(Registry.java:133)
	at net.sf.fmj.registry.Registry.getInstance(Registry.java:105)
	at javax.media.pm.PackageManager.<clinit>(PackageManager.java:23)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:375)
	at javax.media.PackageManager.init(PackageManager.java:117)
	at javax.media.PackageManager.getProtocolPrefixList(PackageManager.java:93)
	at org.jitsi.impl.neomedia.codec.FMJPlugInConfiguration.registerCustomPackages(FMJPlugInConfiguration.java:332)
	at org.jitsi.impl.neomedia.MediaServiceImpl.setupFMJ(MediaServiceImpl.java:1629)
	at org.jitsi.impl.neomedia.MediaServiceImpl.<clinit>(MediaServiceImpl.java:254)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:375)
	at org.jitsi.impl.libjitsi.LibJitsiImpl$ServiceLock.initializeService(LibJitsiImpl.java:173)
	... 23 more
2024-09-04 19:08:12.016 SEVERE: [18] net.java.sip.communicator.util.osgi.DependentActivator.addingService: Failed to start bundle with services
java.lang.IllegalArgumentException: service
	at org.jitsi.impl.osgi.framework.launch.FrameworkImpl.registerService(FrameworkImpl.java:329)
	at org.jitsi.impl.osgi.framework.BundleContextImpl.registerService(BundleContextImpl.java:225)
	at net.java.sip.communicator.impl.neomedia.NeomediaActivator.startWithServices(NeomediaActivator.java:419)
	at net.java.sip.communicator.util.osgi.DependentActivator.addingService(DependentActivator.java:103)
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:943)
	at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:871)
	at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	at org.osgi.util.tracker.AbstractTracked.trackInitial(AbstractTracked.java:183)
	at org.osgi.util.tracker.ServiceTracker.open(ServiceTracker.java:321)
	at org.osgi.util.tracker.ServiceTracker.open(ServiceTracker.java:264)
	at net.java.sip.communicator.util.osgi.DependentActivator.start(DependentActivator.java:79)
	at org.jitsi.impl.osgi.framework.BundleImpl.start(BundleImpl.java:428)
	at org.jitsi.impl.osgi.framework.launch.FrameworkImpl.startLevelChanged(FrameworkImpl.java:483)
	at org.jitsi.impl.osgi.framework.startlevel.FrameworkStartLevelImpl$Command.run(FrameworkStartLevelImpl.java:133)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

File: jigasi/Dockerfile (hunk 1 above)
Comment (damencho) [reply jigasi/Dockerfile]: there is an error with headless one for jigasi, because of the renderers in the media impl. I will try these days debugging it.

**Pull request merged**